bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/device/TapoSmartBulb.java

```diff
@@ -0,0 +1,169 @@
+                case CHANNEL_COLOR:
+                    if (command instanceof HSBType) {
+                        setColor((HSBType) command);
+                        refreshInfo = true;
+                    }
+                    break;
+                default:
+                    logger.warn("({}) command type '{}' not supported for channel '{}'", uid, command.toString(),
+                            channelUID.getId());
+            }
+        }
+
+        /* refreshInfo */
+        if (refreshInfo) {
+            queryDeviceInfo();
+        }
+    }
+
+    /**
+     * SET BRIGHTNESS
+     * 
+     * @param newBrightness percentage 0-100 of new brightness
+     */
+    protected void setBrightness(Integer newBrightness) {
+        /* switch off if 0 */
+        if (newBrightness == 0) {
+            connector.sendDeviceCommand("device_on", false);
+        } else {
+            HashMap<String, Object> newState = new HashMap<>();
+            newState.put("device_on", true);
+            newState.put("brightness", newBrightness);
+            connector.sendDeviceCommands(newState);
+        }
+    }
+
+    /**
+     * SET COLOR
+     * 
+     * @param command
+     */
+    protected void setColor(HSBType command) {
+        HashMap<String, Object> newState = new HashMap<>();
+        newState.put("device_on", true);
+        newState.put("hue", command.getHue());
+        newState.put("saturation", command.getSaturation());
+        newState.put("brightness", command.getBrightness());
+        connector.sendDeviceCommands(newState);
+    }
+
+    /**
+     * SET COLORTEMP
+     * 
+     * @param colorTemp (Integer) in Kelvin
+     */
+    protected void setColorTemp(Integer colorTemp) {
+        HashMap<String, Object> newState = new HashMap<>();
+        colorTemp = limitVal(colorTemp, BULB_MIN_COLORTEMP, BULB_MAX_COLORTEMP);
+        newState.put("device_on", true);
+        newState.put("color_temp", colorTemp);
+        connector.sendDeviceCommands(newState);
+    }
+
+    /**
+     * UPDATE PROPERTIES
+     * 
+     * @param TapoDeviceInfo
+     */
+    @Override
+    protected void devicePropertiesChanged(TapoDeviceInfo deviceInfo) {
+        super.devicePropertiesChanged(deviceInfo);
+        publishState(getChannelID(CHANNEL_GROUP_ACTUATOR, CHANNEL_OUTPUT), getOnOffType(deviceInfo.isOn()));
+        publishState(getChannelID(CHANNEL_GROUP_ACTUATOR, CHANNEL_BRIGHTNESS),
+                getPercentType(deviceInfo.getBrightness()));
+        publishState(getChannelID(CHANNEL_GROUP_ACTUATOR, CHANNEL_COLOR_TEMP),
+                getDecimalType(deviceInfo.getColorTemp()));
+        publishState(getChannelID(CHANNEL_GROUP_ACTUATOR, CHANNEL_COLOR), deviceInfo.getHSB());
+
+        publishState(getChannelID(CHANNEL_GROUP_DEVICE, CHANNEL_WIFI_STRENGTH),
+                getDecimalType(deviceInfo.getSignalLevel()));
+        publishState(getChannelID(CHANNEL_GROUP_DEVICE, CHANNEL_ONTIME), getDecimalType(deviceInfo.getOnTime()));
```

Comment:
If you use Units of Measure like `Number:Time`, this must be of type `QuantityType`.